libs/oracledb/langchain_oracledb/vectorstores/oraclevs.py

```diff
@@ -0,0 +1,1929 @@
+                "The embedding_function is neither Embeddings nor callable."
+            )
+
+    def _embed_query(self, text: str) -> List[float]:
+        if isinstance(self.embedding_function, Embeddings):
+            return self.embedding_function.embed_query(text)
+        else:
+            return self.embedding_function(text)
+
+    async def _aembed_query(self, text: str) -> List[float]:
+        if isinstance(self.embedding_function, Embeddings):
+            return await self.embedding_function.aembed_query(text)
+        elif inspect.isawaitable(self.embedding_function):
+            return await self.embedding_function(text)
+        else:
+            return self.embedding_function(text)
+
+    @_handle_exceptions
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[Dict[Any, Any]]] = None,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Add more texts to the vectorstore index.
+        Args:
+          texts: Iterable of strings to add to the vectorstore.
+          metadatas: Optional list of metadatas associated with the texts.
+          ids: Optional list of ids for the texts that are being added to
+          the vector store.
+          kwargs: vectorstore specific parameters
+        """
+
+        texts = list(texts)
+        processed_ids = get_processed_ids(texts, metadatas, ids)
+
+        embeddings = self._embed_documents(texts)
+        if not metadatas:
+            metadatas = [{} for _ in texts]
+
+        docs: List[Tuple[Any, Any, Any, Any]]
+        if self.insert_mode == "clob":
+            docs = [
+                (
+                    id_,
+                    json.dumps(embedding),
+                    json.dumps(metadata)
+                    if self.json_insert_mode != "json"
+                    else metadata,
+                    text,
+                )
+                for id_, embedding, metadata, text in zip(
+                    processed_ids, embeddings, metadatas, texts
+                )
+            ]
+        else:
+            docs = [
+                (
+                    id_,
+                    array.array("f", embedding),
+                    json.dumps(metadata)
+                    if self.json_insert_mode != "json"
+                    else metadata,
+                    text,
+                )
+                for id_, embedding, metadata, text in zip(
+                    processed_ids, embeddings, metadatas, texts
+                )
+            ]
+
+        connection = _get_connection(self.client)
+        if connection is None:
+            raise ValueError("Failed to acquire a connection.")
+        with connection.cursor() as cursor:
+            if self.json_insert_mode == "json":
+                cursor.setinputsizes(None, None, self.json_type, None)
+            cursor.executemany(
+                f"INSERT INTO {self.table_name} (id, embedding, metadata, "
```

Comment:
Maybe because I haven't stepped through the code in a debugger, but it seems strange to me the code isn't working with the VECTOR datatype for embeddings.

Reply:
I did not understand the comment completely, but I removed the insert_mode since we only want to support new versions. Do you think the new version check is what we should have?

Reply:
we use json.dumps(metadata) to convert to string. we can directly pass JSON value by using DB_TYPE_JSON  .
`
https://python-oracledb.readthedocs.io/en/latest/user_guide/json_data_type.html
`


If we convert to string in JSON column, then we loose benefits of JSON text search etc (when we do hybrid search later... )

Reply:
Since you passed `array.array("f", embedding),` DB_TYPE_VECTOR is not needed. client side vector image is constructed.

Reply:
`json.dumps(metadata)` was coming from the earlier implementation where we wanted to support older versions. It is now removed from the code as we require new versions in this library.

Reply:
Thanks